applications/StructuralMechanicsApplication/custom_constitutive/linear_plane_strain.cpp

```diff
@@ -207,38 +84,76 @@ void LinearPlaneStrain::GetLawFeatures(Features& rFeatures)
 
     //Set the spacedimension
     rFeatures.mSpaceDimension = 2;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStrain::CalculateElasticMatrix(Matrix& C, ConstitutiveLaw::Parameters& rValues)
+{
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const double& E = r_material_properties[YOUNG_MODULUS];
+    const double& NU = r_material_properties[POISSON_RATIO];
 
+    this->CheckClearElasticMatrix(C);
+
+    const double c0 = E / ((1.00 + NU)*(1 - 2 * NU));
+    const double c1 = (1.00 - NU)*c0;
+    const double c2 = c0 * NU;
+    const double c3 = (0.5 - NU)*c0;
+
+    C(0, 0) = c1;
+    C(0, 1) = c2;
+    C(1, 0) = c2;
+    C(1, 1) = c1;
+    C(2, 2) = c3;
 }
 
 //************************************************************************************
 //************************************************************************************
 
-int LinearPlaneStrain::Check(const Properties& rMaterialProperties,
-                             const GeometryType& rElementGeometry,
-                             const ProcessInfo& rCurrentProcessInfo)
+void LinearPlaneStrain::CalculatePK2Stress(
+    const Vector& rStrainVector,
+    Vector& rStressVector,
+    ConstitutiveLaw::Parameters& rValues
+)
 {
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const double& E = r_material_properties[YOUNG_MODULUS];
+    const double& NU = r_material_properties[POISSON_RATIO];
+
+    const double c0 = E / ((1.00 + NU)*(1 - 2 * NU));
+    const double c1 = (1.00 - NU)*c0;
+    const double c2 = c0 * NU;
+    const double c3 = (0.5 - NU)*c0;
+
+    rStressVector[0] = c1 * rStrainVector[0] + c2 * rStrainVector[1];
+    rStressVector[1] = c2 * rStrainVector[0] + c1 * rStrainVector[1];
+    rStressVector[2] = c3 * rStrainVector[2];
+}
 
-    if(YOUNG_MODULUS.Key() == 0 || rMaterialProperties[YOUNG_MODULUS]<= 0.00)
-    {
-        KRATOS_ERROR << "YOUNG_MODULUS has Key zero or invalid value " << std::endl;
-    }
+//************************************************************************************
+//************************************************************************************
 
-    const double& nu = rMaterialProperties[POISSON_RATIO];
-    const bool check = bool( (nu >0.499 && nu<0.501 ) || (nu < -0.999 && nu > -1.01 ) );
+void LinearPlaneStrain::CalculateCauchyGreenStrain(Parameters& rValues, Vector& rStrainVector)
+{
+    //1.-Compute total deformation gradient
+    const Matrix& F = rValues.GetDeformationGradientF();
 
-    if(POISSON_RATIO.Key() == 0 || check==true)
-    {
-        KRATOS_ERROR << "POISSON_RATIO has Key zero invalid value " << std::endl;
-    }
+    // for shells/membranes in case the DeformationGradient is of size 3x3
```

Comment:
@loumalouomega I think this check is missing in the hyperelastic laws you did, so this is a reminder